### DISCUSSION TITLE
Custom Borg Backup of separate folders (how to start a custom Borg instance from within the container?)

### DISCUSSION THREAD

**Discussion by DonkeeeyKong** [nextcloud/all-in-one#4982] [Questions] [answered]
Hello,

Many thanks to everybody involved for this amazing product and also for the nice community! Quick question:

I have several non-Nextcloud folders that I backup using Borg. Adding them to the automatic Nextcloud backup is not the best solution for me because they all get merged into the same archive. I'd rather have separate archives with separate tags for each source. (Edit/Addition: More importantly: I need to exclude several sub-directories which is not supported by the Nextcloud-AIO GUI.)

So I have been trying with using Borg Backup directly on the server and setting the Nextcloud repository as a destination. I use a `aio-lockfile` as is recommended in the documentation and the backup seems to work fine and I profit a lot from the deduplication compared to using a different repository as I have been doing earlier. 

However, now the Nextcloud backup always has to synchronize the chunks index before starting. From what I read this is not a big problem, though not ideal. 
Apparently the cache is always stored in the home directory of the user running borg, in `~/.cache/borg`. I believe if I started the backups from within the container this problem would go away. 

Is that right? Does anybody know how to do this?

Many thanks in advance!

**Comment by szaimen** [comment]
Hi, maybe this helps? https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-stopstartupdate-containers-or-trigger-the-daily-backup-from-a-script-externally

**Reply by DonkeeeyKong** [reply]
Hi. Thanks a lot for the reply! That's not quite what I meant though. I am using this script for my regular Nextcloud backups, but the script doesn't support custom calls of Borg (i.e. use custom tag/archive names for external folders, exclude specific subdirecories, etc.).
What I am trying to find out is whether it's possible to directly start borg with custom options from within the container to backup additional folders to the existing Nextcloud repository. And if it's possible how it could be done.

**Reply by szaimen** [reply]
We have this, not sure if it solves your use case?
![image](https://github.com/user-attachments/assets/45f797c9-9439-48fa-b458-fc1f522886f6)

**Reply by DonkeeeyKong** [reply]
Thank you for your help! The problem is that I can't exclude specific subdirectories with this. If I start borg directly I can just use the `--exclude` flag for that (and use custom archive names instead of having everything merged into the `*-additional-host-mounts` tag, but that's more a cosmetic preference). 

If it's not possible to start borg with a custom command from within the container I might just have to continue doing what I am doing now (using borg on the server for my additional folders and setting the nextcloud repository as destination) - and live with the chunks index getting out of sync. Or backup to a different repository and lose the deduplication benefits. :) Not a big problem. Thanks for your help and all of your work!

**Reply by szaimen** [reply]
Maybe this helps? https://github.com/nextcloud/all-in-one/blob/e1b5ba884f68ef9bc14b1051bc72d97ff133637c/Containers/borgbackup/backupscript.sh#L152

**Reply by szaimen** [reply]
See https://borgbackup.readthedocs.io/en/stable/usage/create.html

**Reply by DonkeeeyKong** [reply]
> Maybe this helps?
> 
> https://github.com/nextcloud/all-in-one/blob/e1b5ba884f68ef9bc14b1051bc72d97ff133637c/Containers/borgbackup/backupscript.sh#L152

I am sorry. I don't understand what you mean. Do you suggest I edit that script and add my exclude wishes?

**Reply by szaimen** [reply, marked as answer]
No, see 
![Screenshot_20240717_181257_Gallery](https://github.com/user-attachments/assets/2a9534b9-26ac-41fa-9bc6-f683a3ddfecc)

**Reply by DonkeeeyKong** [reply]
Ahh. You suggest to add a CACHEDIR.TAG to the directories I want to exclude? That's actually a good idea. Sorry, took me some time to understand what you were trying to say...

**Reply by DonkeeeyKong** [reply]
Just tried it. This works. Thank you!
